### ISSUE TITLE
Style/VariableName doesn't seem to work with method parameters

### ISSUE THREAD

**Issue by mateusmedeiros** [bbatsov/rubocop#2335]
I don't know if this is intended behaviour, but if it is, I think it should be changed (unless there's a specific cop for that, in which case I'd really like to know which it is, because I couldn't find any).

If I have the following method in a ruby script:
```ruby
def downloadAnother(urlFileHash)
  variableWithCamelCase = 2
  urlFileHash.each do |url, file|
    puts "downloading #{url} to #{file}"
  end
end
```

Rubocop will report 3 offenses from this method:
```
C: Use snake_case for method names.
def downloadAnother(urlFileHash)
    ^^^^^^^^^^^^^^^
```
```
W: Useless assignment to variable - variableWithCamelCase.
  variableWithCamelCase = 2
  ^^^^^^^^^^^^^^^^^^^^^
```
```
C: Use snake_case for variable names.
  variableWithCamelCase = 2
  ^^^^^^^^^^^^^^^^^^^^^
```

But I was expecting `Style/VariableName` to also detect offenses in parameters names, like this:
```
C: Use snake_case for variable names.
def downloadAnother(urlFileHash)
                    ^^^^^^^^^^^
```

<br>
This is the original method I had for which rubocop is reporting no offense at all:
```ruby
def download(urlFileHash)
  urlFileHash.each do |url, file|
    puts "Downloading #{url} to the file #{file}..."
    File.open(file, 'wb') do |saved_file|
      open(url, 'rb') do |read_file|
        saved_file.write(read_file.read)
      end
    end
  end
end
```

I installed rubocop straight from master and it still reports no offense.
Here is my `rubocop -V`:
```
0.34.2 (using Parser 2.2.3.0, running on ruby 2.2.3 x86_64-linux)
```

Thanks in advance.

**Comment by jonas054** [comment]
Makes sense. Let `Style/VariableName` check parameter names too.

**Comment by kjvarga** [comment]
This still isn't catching bad variable names in arguments passed to a method, and it doesn't catch bad `let` variable definitions (thought I think that one is understandable).  It _does_ catch bad parameters in a method definition, and bad variable assignments.

**Comment by kjvarga** [comment]
Created issue: https://github.com/rubocop-hq/rubocop/issues/7275
